extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/HibernateValidatorRecorder.java

```diff
@@ -75,10 +76,11 @@ public void created(BeanContainer container) {
                     configuration.localeResolver(localeResolver);
                 }
 
-                configuration
-                        .builtinConstraints(detectedBuiltinConstraints)
+                configuration.builtinConstraints(detectedBuiltinConstraints)
                         .initializeBeanMetaData(classesToBeValidated)
-                        .locales(localesBuildTimeConfig.locales)
+                        // Locales, Locale ROOT means all locales in this setting.
+                        .locales(localesBuildTimeConfig.locales.contains(Locale.ROOT) ? Set.of(Locale.getAvailableLocales())
+                                : localesBuildTimeConfig.locales)
```

Comment:
Hmmm, what's the rationale of that? Can't we rely on the config being configured to `all` instead?

Reply:
@gsmet It covers situation when user mis/configures something like: `quarkus.locales=fr,all,en-GB` as ROOT is what I use in place of "all".

If you find it clumsy, I can redefine that, check config instead for string "all"  and treat anything like the aforementioned configuration as an error in LocaleProcessor in core.

Reply:
And so there's no way for the locale to be defined as root by the user/app? Because if you use it as marker, we absolutely need to never be used by a user/app.

Reply:
@gsmet  https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html#ROOT

That's what I was working with. I.e. user would have to set empty string for the property quarkus.locales to be parsed as ROOT. Setting empty string is not allowed by our value converter:

```
Failed to build quarkus application: SRCFG00040: The config property quarkus.locales is defined as the empty String ("") which the following Converter considered to be null: io.smallrye.config.Converters$CollectionConverter
```

Reply:
I suppose we could live with it. I tried to use `LANG=C` and AFAICS, I end up with an English locale.

Reply:
yes, it all falls back to ENGLISH

Reply:
i.e. tests fail due to this, but I already documented it here:  https://github.com/quarkusio/quarkus/blob/main/integration-tests/locales/src/test/java/io/quarkus/locales/it/LocalesIT.java#L17